tests/scenario/test_deploy.py

```diff
@@ -54,7 +53,9 @@ def test_config_changed_cannot_connect():
 
 
 def test_deploy_and_set_alerts_error_scenario():
-    MimirK8SOperatorCharm._set_alerts = Mock(side_effect=PebbleError)
+    MimirK8SOperatorCharm._set_alerts = Mock(
+        side_effect=BlockedStatusError("Failed to push updated alert files; see debug logs")
+    )
```

Comment:
- Asserting on `BlockedStatusError` would be whitebox testing whereas `PebbleError`, `ChangeError` etc. would be blackbox testing.
- Could use `pytest.parametrize` to repeat the test with all of them.

Reply:
We are not asserting on `BlockedStatusError` we are asserting on `BlockedStatus`... We are mocking `MimirK8SOperatorCharm._set_alerts` with `BlockedStatusError` Exception (that is raised when `PebbleError`, `ProtocolError` is catched).

The idea of this test is to mock `_set_alerts` raising a custom Exception that afterwards is catched in the `_configure` method